R/convert_RCMIP.R

```diff
@@ -20,42 +22,44 @@ convert_rcmipCMIP6_hector <- function(scenario = NULL){
 
   # Make sure data exists for the scenario(s) selected to process. 
   data_scns <- unique(emiss_data$Scenario, conc_data$Scenario)
-  missing   <- !scenario %in% data_scns
-  assertthat::assert_that(sum(missing) == 0,
-                          msg = paste0('The following scenarios cannot be processed: ', paste(scenarios[missing], collapse = ', ')))
+  available <- scenario %in% data_scns
+  assert_that(all(available), msg = paste0('The following scenarios cannot be processed: ', paste(scenarios[!available], collapse = ', ')))
 
 
-  # Determine the columns that contain identifier information, such as the model, scneairo, region, variable,
+  # Determine the columns that contain identifier information, such as the model, scenario, region, variable,
   # unit, ect. These columns will be used to transform the data from being wide to long so that each row
-  # corresponds to concenration for a specific year.
-  id_vars <- which(!grepl(pattern = "[[:digit:]]{4}", x = names(conc_data)))
-  conc_long <- data.table::melt.data.table(data = conc_data, id.vars = id_vars,
+  # corresponds to concentration for a specific year.
+  id_vars <- which(!grepl(pattern = "^X[[:digit:]]{4}", x = names(conc_data)))
+  conc_long <- melt.data.table(data = conc_data, id.vars = id_vars,
                                            variable.name = "year", value.name = "value",
                                            variable.factor = FALSE)
   
 
 
-  id_vars <- which(!grepl(pattern = "[[:digit:]]{4}", x = names(emiss_data)))
-  emiss_long <- data.table::melt.data.table(data = emiss_data, id.vars = id_vars,
+  id_vars <- which(!grepl(pattern = "^X[[:digit:]]{4}", x = names(emiss_data)))
+  emiss_long <- melt.data.table(data = emiss_data, id.vars = id_vars,
                                             variable.name = "year", value.name = "value",
                                             variable.factor = FALSE)
 
-  # Concatenate the long emissions and concetnration data tables together and subset so that
-  # only the scenarios of intrest will be converted. Remove the NA entries that arose when converted from
+  # Concatenate the long emissions and concentration data tables together and subset so that
+  # only the scenarios of interest will be converted. Remove the NA entries that were added when converting from
   # the wide to long format.
   raw_inputs <- stats::na.omit(rbind(emiss_long, conc_long))
   raw_inputs <- raw_inputs[Scenario %in% scenario  & Region == "World"][ , year :=  as.integer(substr(year, 2, 5))]
-  raw_inputs <- remove_spaces(df = raw_inputs, cols = c("Variable", "Mip_Era", "Model", "Scenario", "Region"))
-  
-  
   
+  # Remove trailing spaces from the RCMIP inputs. 
+  cols_to_modify <- which(names(raw_inputs) %in% c("Model", "Scenario", "Region", "Variable", "Unit", "Mip_Era"))
```

Comment:
Will these column names ever change?  If so, then they should be passed into the function or read in from a YAML file.  This comment applies to all hard-coded names thereafter and `expected_year` value that on line 81 that could be a default.

Reply:
They shouldn't ever change, they are really only relevant to the RCMIP files because of some funky formatting.